src/test_payload_dataset_manager.cc

```diff
@@ -46,54 +46,47 @@ TEST_CASE("PayloadDatasetManager::GroupPayloadsBySession")
 {
   SUBCASE("valid input data")
   {
-    auto model_parser{std::make_shared<ModelParser>()};
+    const std::string name{"session_id"};
+    const std::string datatype{"BYTES"};
+    const std::vector<int64_t> shape{{1}};
+    const bool is_shape_tensor{false};
+    const bool is_optional{false};
+
+    const ModelTensor session_id_model_tensor{
+        name, datatype, shape, is_shape_tensor, is_optional};
 
-    const std::string model_name{""};
-    const std::string model_version{""};
-    const int32_t batch_size{0};
-    const InferenceLoadMode inference_load_mode{
-        InferenceLoadMode::SessionConcurrency};
+    auto model_parser{std::make_shared<ModelParser>()};
 
-    model_parser->InitOpenAI(
-        model_name, model_version, batch_size, inference_load_mode);
+    (*model_parser->Inputs())[name] = session_id_model_tensor;
```

Comment:
Cleaned up some random stuff in this test. Skipped calling `InitOpenAI()` in favor of just creating the `ModelTensor` itself and assigning to a blank `ModelParser` to be used later in the test.

Previously, the test was relying on InitOpenAI() to create `ModelTensor`s and assign them to the `ModelParser`.

This just relies on less code now, further isolating what we want to test.